here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/WriteFeatureApiTask.java

```diff
@@ -228,6 +234,200 @@ protected void init() {}
+          // This function was then misused somewhere. FIND AND FIX IT!!
+          return returnError(
+              XyzError.EXCEPTION,
+              "Internal server error.",
+              "Unsupported HttpResponseType was called: {}",
+              responseType);
+        }
+        // Else none of the features exists in storage, will create them later
+      }
+      // Attempt patching, keeping the order of the features from the request
+      patchedFeatures = performInMemoryPatching(featuresFromRequest, featuresToPatchFromStorage);
+      // As applicable, modify features based on parameters supplied
+      for (XyzFeature featureFromRequest : patchedFeatures) {
+        addTagsToFeature(featureFromRequest, addTags);
+        removeTagsFromFeature(featureFromRequest, removeTags);
+      }
+    }
+    final WriteXyzFeatures wrRequest = RequestHelper.upsertFeaturesRequest(spaceId, patchedFeatures);
+    // Forward request to NH Space Storage writer instance
+    try (final IWriteSession writer = naksha().getSpaceStorage().newWriteSession(context(), true)) {
+      final Result wrResult = writer.execute(wrRequest);
+      if (wrResult == null) {
+        // unexpected null response
+        writer.rollback(true);
+        writer.close();
+        return returnError(
+            XyzError.EXCEPTION,
+            "Unexpected null result.",
+            "Received null result after writing patched features, rolled back.");
+      } else if (wrResult instanceof ErrorResult er) {
+        writer.rollback(true);
+        writer.close();
+        try (ForwardCursor<XyzFeature, XyzFeatureCodec> resultCursor = er.getXyzFeatureCursor()) {
+          if (!resultCursor.hasNext()) {
+            throw new NoSuchElementException("Error Result Cursor is empty");
+          }
+          while (resultCursor.hasNext()) {
+            if (!resultCursor.next()) {
+              throw new RuntimeException("Unexpected invalid error result");
+            }
+            // Check if there is an error that is not about mismatching UUID
+            if (EExecutedOp.ERROR.equals(resultCursor.getOp())) {
+              if (XyzError.CONFLICT
+                  .equals(Objects.requireNonNull(resultCursor.getError())
+                          .err)) {
+                // Other types of error, will not retry
+                return returnError(
+                    resultCursor.getError().err,
+                    resultCursor.getError().msg,
+                    "Received error result {}",
+                    resultCursor.getError());
+              }
+              // Else it was because of UUID mismatched
+              final String featureIdFromErr = resultCursor.getId();
+              // Find the requested change for the corresponding feature with that ID
+              for (XyzFeature requestedChange : featuresFromRequest) {
+                if (requestedChange.getId().equals(featureIdFromErr)) {
```

Comment:
I suggest reversing this condition assuming `requestedChange.getId()` can be "by design" null ... eventhough it is not null as per present implementation. So, it should be `featureIdFromErr.equals(requestedChange.getId())`